web/src/message_view.ts

```diff
@@ -767,6 +776,89 @@ export let show = (raw_terms: NarrowTerm[], show_opts: ShowMessageViewOpts): voi
             then_select_offset,
         );
 
+        if (id_info.first_unread_msg_id_pending_server_verification) {
+            const params = message_fetch.get_parameters_for_message_fetch_api({
+                anchor: "first_unread",
+                num_before: 0,
+                num_after: 0,
+                cont() {
+                    // ignore
+                },
+                msg_list_data: msg_list.data,
+            });
+            void channel.get({
+                url: "/json/messages",
+                data: params,
+                success(raw_data) {
+                    // If we switched narrow, there is nothing to do.
+                    if (
+                        msg_list !== message_lists.current ||
+                        !id_info.first_unread_msg_id_pending_server_verification
+                    ) {
+                        return;
+                    }
```

Comment:
Hmm, is it possible we want to store this state on the `MessageListData` instead of the transient `id_info`, so that the state survives the Back button properly? In that case, we'd not want to return here.

Reply:
I think that would make this code non-intuitive. Also, we would have another fetch request to verify the first unread when user presses back button.

Reply:
In the case that we found no older unreads, I believe that state should be safely correct for the duration of the browser window. New mark-as-read operations will always be fully transmitted to clients, so we can only be missing data on those that predate the current browser window opening.

But maybe there's a corner case involving a topic with very old unreads being moved/merged into the current view -- is that what you're worried about?

Reply:
Thinking about this agian, I guess it is simply not reliable to rely on the verified first unread message id for `MessageListData` when user presses the back button since user could have another window open where they could mark more messages unread.

Reply:
If another window does that, though, won't we get an event and thus update our local store to have those unreads? Thus, if one of them is oldest, we should be all set? IDK, maybe it's an optimization we don't need, since it's messy and complex.